firefox-ios/Client/Frontend/Browser/SearchViewController.swift

```diff
@@ -848,7 +856,30 @@ class SearchViewController: SiteTableViewController,
             break
         }
     }
-}
+
+    private func filterData(data: Cursor<Site>) -> [Site?] {
+        return data.filter { item in
+            guard profile.prefs.boolForKey(PrefsKeys.FirefoxSuggestShowSponsoredSuggestions) ?? false  else { return true }
+
+            guard let urlString = item?.url as? String,
+                  let searchParamString = profile.prefs.stringForKey(PrefsKeys.FirefoxSuggestFilterSearchParameter),
+                  let url = URL(string: urlString) else {return true}
```

Comment:
```suggestion
                  let components = URLComponents(string: urlString) else { return true }
```

I think you could use `URLComponents` directly here—no need to turn it into a `URL` just to turn it into a `URLComponents` in `containsQueryParameters`.

Reply:
Ahhh, I'm sorry, I forgot that you're also filtering tabs, and those have a `URL` 😅)